crates/re_ui/src/list_item.rs

```diff
@@ -287,9 +298,20 @@ impl<'a> ListItem<'a> {
     }
 
     /// Draw the item.
-    pub fn show(self, ui: &mut Ui) -> Response {
+    pub fn show(self, ui: &mut Ui, indent_mode: IndentMode) -> Response {
```

Comment:
I'm curious - why did you chose to make this an argument to `show` instead of a property of `ListItem`? Is it so we consider it carefully wherever we have a list item?

Reply:
Spot on. It's way too easy to forget about it otherwise. IMHO, the alternative would be to have 3 distinct APIs, eg.:
- `show_hierarchical_with_content()`
- `show_hierarchical()`
- `show_flat()`

Maybe that would actually be better?

Reply:
(Also, making it a property would be a little weird with the existing `show_collapsing`, as it would be ignored in that case.)

Reply:
`show_hierarchical` etc would certainly be more concise, and could be a nice addition

Reply:
Right, I'll make that change. I like it better too.